Plugin/Model/OrderRepositoryPlugin.php

```diff
@@ -0,0 +1,119 @@
+     */
+    private $sourcesRepository;
+
+    /**
+     * @var OrderExtensionFactory
+     */
+    private $orderExtensionFactory;
+
+    /**
+     * @var SourceSelectionResultInterfaceFactory
+     */
+    private $sourceSelectionResultInterfaceFactory;
+
+    /**
+     * @var SourcesConverter
+     */
+    private $sourcesConverter;
+
+    /**
+     * OrderRepositoryPlugin constructor.
+     * @param SourcesRepository $sourcesRepository
+     * @param OrderExtensionFactory $orderExtensionFactory
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultInterfaceFactory
+     * @param SourcesConverter $sourcesConverter
+     */
+    public function __construct(
+        SourcesRepository $sourcesRepository,
+        OrderExtensionFactory $orderExtensionFactory,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultInterfaceFactory,
+        SourcesConverter $sourcesConverter
+    ) {
+        $this->sourcesRepository = $sourcesRepository;
+        $this->orderExtensionFactory = $orderExtensionFactory;
+        $this->sourceSelectionResultInterfaceFactory = $sourceSelectionResultInterfaceFactory;
+        $this->sourcesConverter = $sourcesConverter;
+    }
+
+    /**
+     * @param OrderRepository $subject
+     * @param OrderInterface $result
+     * @return OrderInterface
+     */
+    public function afterGet(OrderRepository $subject, OrderInterface $result): OrderInterface
+    {
+        return $this->applyExtensionAttributesToOrder($result, null);
+    }
+
+    /**
+     * @param OrderRepository $subject
+     * @param OrderSearchResultInterface $result
+     * @return OrderSearchResultInterface
+     */
+    public function afterGetList(OrderRepository $subject, OrderSearchResultInterface $result): OrderSearchResultInterface
+    {
+        $sourcesCollection = $this->sourcesRepository->getOrdersSourcesCollection();
+
+        foreach ($result->getItems() as $item) {
+            $sourceSelectionItems = $this->sourcesConverter
+                ->convertSourcesArrayToSourceSelectionItems($sourcesCollection[$item->getEntityId()]);
+
+            $sourceSelectionResult = $this->sourceSelectionResultInterfaceFactory->create([
+                'sourceItemSelections' => $sourceSelectionItems,
+                'isShippable' => true
+            ]);
+
+            $this->applyExtensionAttributesToOrder($item, $sourceSelectionResult);
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param OrderInterface $order
+     * @param SourceSelectionResultInterface|null $sourceSelectionResult
+     * @return OrderInterface
+     */
+    private function applyExtensionAttributesToOrder(
+        OrderInterface $order,
+        SourceSelectionResultInterface $sourceSelectionResult = null
+    ): OrderInterface
+    {
+        if ($sourceSelectionResult === null) {
+            $sourceSelectionResult = $this->sourcesRepository->getSourceSelectionResultByOrderId($order->getEntityId());
```

Comment:
Again, this will be calling load within a loop https://github.com/AmpersandHQ/magento2-disable-stock-reservation/pull/6/files#diff-b94e3006e0b587e8e847eb7646100f53R87